analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.FlowCaptureOperation.cs

```diff
@@ -0,0 +1,354 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+extern alias csharp;
+namespace SonarAnalyzer.Test.LiveVariableAnalysis;
+
+public partial class RoslynLiveVariableAnalysisTest
+{
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingAssignment()
+    {
+        /*      Block 1
+        *     #0=stringParam
+        *          |
+        *       Block 2
+        *        #1=#0
+        *      #1 is null --+
+        *          |        |
+        *       Block 3     |
+        *       #0=Hello    |
+        *          |        |
+        *        Exit <-----+
+        */
+        const string code = """stringParam ??= "Hello";""";
+        var context = CreateContextCS(code, additionalParameters: "string stringParam");
+        context.ValidateEntry(LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("stringParam"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingOperator()
+    {
+        /*       Block 1
+         *    #0=stringParam
+         *     #0 is null
+         *        /   \
+         *       /     \
+         *   Block 2  Block 3
+         *   #1=#0  #1="Hello"
+         *       \     /
+         *       Block 4
+         *         s=#1
+         *          |
+         *        Exit
+         */
+        const string code = """string s = stringParam ?? "Hello";""";
+        var context = CreateContextCS(code, additionalParameters: "string stringParam");
+        context.ValidateEntry(LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("stringParam"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullConditionalOperators()
+    {
+        /*       Block 1
+         *    #1=stringParam
+         *     #1 is null
+         *        /   \
+         *       F     T
+         *       /     \
+         *   Block 2  Block 3
+         *  #0=Length #0=stringParam (DefaultValueOperation)
+         *       \     /
+         *       Block 4
+         *      result=#1
+         *          |
+         *        Exit
+         */
+        const string code = """int? anInt =  stringParam?.Length;""";
```

Comment:
To be aligned with the graph above `result=#1`
```suggestion
        const string code = """int? result =  stringParam?.Length;""";
```